### PR TITLE
Tl 64

### LINKED ISSUE #64: [Feature] Extend calendar view functionality
**Is your feature request related to a problem? Please describe.**
Right now(as implemented in #15 , the calendar view shows only a badge with the number of todos of each day.

**Describe the solution you'd like**
Extend UI functionality and make the badge clickable. 
When the user clicks that badge, open a pop-up or append below the calendar a view to show details about those todos.

### PR DESCRIPTION
**Pull request by VBorisof**
- Added CalendarTodoViewModel to use in the calendar view.
- Added a modal to display the todos for the day.

### CODE REVIEW

File: TodoList.Web/Controllers/TodosController.cs [hunk 1]
```diff
@@ -39,13 +40,30 @@ public async Task<IActionResult> Home()
...
+                        DueToDay = t.DueTo.ToDateTimeOffset().Day
+                    })
+                .GroupBy(t => 
+                    t.DueToDay
+                )
+                .ToDictionary(
+                    keySelector: g => g.Key,
+                    elementSelector: g => g.Select(t => t)
+                );
+
```
Comment (amoraitis) [inline TodoList.Web/Controllers/TodosController.cs:61]: Can you please move it to the TodoItemService? The controller would remain thinner that way. Also `monthlyItems` is not used any more, feel free to modify+rename the service method `ITodoItemService.GetMonthlyItems(currentUser, month)`

File: TodoList.Web/Views/Todos/Home.cshtml [hunk 2]
```diff
@@ -188,15 +188,45 @@
...
+                                                            </li>
+                                                        }
+                                                    </ul>
+                                                </div>
+                                                <div class="modal-footer">
+                                                    <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                                                </div>
+                                            </div>
+                                        </div>
+                                    </div>
```
Comment (amoraitis) [inline TodoList.Web/Views/Todos/Home.cshtml:225]: This modal can be a separate layout file in the future. I will open a different issue for that. I want to see what other razors can be included in their own .cshtml files.

**Review by amoraitis — CHANGES_REQUESTED** [review]
@VBorisof Amazing modal implementation! I really enjoyed that view. Please see my other comment and after that, we can merge.

File: TodoList.Web/Controllers/TodosController.cs (hunk 1 above)
Comment (amoraitis) [reply TodoList.Web/Controllers/TodosController.cs:61]: Actually not. We don't want to use our View Model class in our service. Sorry.

**Review by amoraitis — APPROVED** [review]

**Comment by VBorisof** [comment]
@amoraitis Hey @amoraitis , thanks for review! I saw you mentioned something about moving some logic to services, but then it was gone, is it still relevant?

**Comment by amoraitis** [comment]
> @amoraitis Hey @amoraitis , thanks for review! I saw you mentioned something about moving some logic to services, but then it was gone, is it still relevant?

No, this would require a data mapper class, so not. Thanks for your contribution to the project!

**Pull request merged**